### PR TITLE
Added dist support for EpochMetric and other similar metrics

### LINKED ISSUE #978: Make EpochMetric work in DDP
## 🚀 Feature

Currently, metrics like `EpochMetric` which store complete history of predictions and targets, are not supporting DDP setting. 
Idea is to perform [gather](https://pytorch.org/docs/stable/distributed.html#torch.distributed.gather) on zero rank process in `compute` method to accumulate all predictions and targets across all the processes and then execute metric function...
We may also need to warn about high memory consumption of this approach.

cc @sdesrozis

### PR DESCRIPTION
**Pull request by vfdev-5**
Fixes #978 

Description:

- Added dist support for EpochMetric with tests
- Blocked by absence of `idist.broadcast`

Check list:
* [ ] New tests are added (if a new feature is added)
* [ ] New doc strings: description and/or example code are in RST format
* [ ] Documentation is updated (if required)

### CODE REVIEW

File: ignite/metrics/epoch_metric.py [hunk 1]
```diff
@@ -113,6 +116,12 @@ def update(self, output: Sequence[torch.Tensor]) -> None:
     def compute(self) -> None:
         _prediction_tensor = torch.cat(self._predictions, dim=0)
         _target_tensor = torch.cat(self._targets, dim=0)
+
+        if not self._is_reduced:
+            _prediction_tensor = idist.all_gather(_prediction_tensor)
+            _target_tensor = idist.all_gather(_target_tensor)
+            self._is_reduced = True
+
         return self.compute_fn(_prediction_tensor, _target_tensor)
```
Comment (vfdev-5) [inline ignite/metrics/epoch_metric.py:125]: To reduce computations on CPU we can apply compute_fn only on a single rank and broadcast the result to all processes.
